conmon/conmon.c

```diff
@@ -550,6 +550,45 @@ static bool read_stdio(int fd, stdpipe_t pipe, bool *eof)
 	}
 }
 
+static void on_sigchld(G_GNUC_UNUSED int signal)
+{
+	raise (SIGUSR1);
+}
+
+static void check_child_processes(GHashTable *pid_to_handler)
+{
+	void (*cb) (GPid, int, gpointer);
+
+        for (;;) {
+          int status;
+          pid_t pid = waitpid(-1, &status, WNOHANG);
+
+          if (pid < 0 && errno == EINTR)
+            continue;
+          if (pid < 0)
+            pexit("Failed to read child process status");
+          if (pid == 0)
+            return;
+
+
+          cb = g_hash_table_lookup(pid_to_handler, &pid);
+          if (cb == NULL)
+            continue;
```

Comment:
this happens only in the case when `waitpid` returns a PID.  This should never get into an endless loop, unless zombie processes are created faster than this loop can manage.